### PR TITLE
Migration to typescript

### PR DESCRIPTION
**Pull request by timon-witt**
At my company, we cloned this repository and changed some things to our needs. Part of that, was migrating it to typescript. I want to share this very migration with you, using this pull request.
It is already tested and running in production and it looks good so far. There should not be any breaking changes.

This pull request is a suggestion, of course, as I don't know your familiarity with typescript. What I can say is, that I could already find and fix a handful of bugs while migrating, which were not easily visible with plain javascript.

Furthermore, I would want to provide another pull request, solving this issue: https://github.com/prebid/prebid-outstream/issues/8. I can, however, only do that, if the typescript migration was merged.

### CODE REVIEW

File: src/Utils.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    const viewport_top = getScrollTop();
+    const viewport_height = screen.availHeight;
+    const viewport_bottom = viewport_top + viewport_height;
+
+    const top = getOffset(element).top;
+
+    // TODO: Fix this assertion. style.height is actually a string.
+    const height = (element.style.height as any) as number;
```
Comment (timon-witt) [inline src/Utils.ts:50]: This is unchanged from the javascript version and should definitely be broken. I fixed it in my solution to #8, using `element.clientHeight;`

**Comment by muuki88** [comment]
@hrgui it would be amazing if this gets merged even if prebid is working on a new implementation

**Comment by hrgui** [comment]
@muuki88 as amazing it would be, I'm not an owner of this repo. I just contributed a correction on a typo. https://github.com/prebid/prebid-outstream/pull/4

**Comment by muuki88** [comment]
Damn 😬

Thanks anyways for the feedback ❤️

**Comment by muuki88** [comment]
@mike-chowla any chance this can be merged?

**Comment by muuki88** [comment]
@patmmccann if there's a chance to get this merged, I would start resolving the merge conflicts :smile:

**Comment by patmmccann** [comment]
will discuss with Mike in video committee

**Comment by patmmccann** [comment]
@muuki88 you have write+ on this repo

**Comment by muuki88** [comment]
Close in favor of #18 18

**Pull request closed without merging**